solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/RankSVMModel.java

```diff
@@ -28,36 +29,62 @@
 
 public class RankSVMModel extends LTRScoringModel {
 
-  protected float[] featureToWeight;
+  protected Float[] featureToWeight;
 
   /** name of the attribute containing the weight of the SVM model **/
   public static final String WEIGHTS_PARAM = "weights";
 
-  public RankSVMModel(String name, List<Feature> features,
+  public static RankSVMModel create(String name, List<Feature> features,
       List<Normalizer> norms,
       String featureStoreName, List<Feature> allFeatures,
       Map<String,Object> params) throws ModelException {
+    final RankSVMModel model = new RankSVMModel(name, features,
+        norms, featureStoreName, allFeatures, params);
+    model.validate();
+    return model;
+  }
+
+  public RankSVMModel(String name, List<Feature> features,
+      List<Normalizer> norms,
+      String featureStoreName, List<Feature> allFeatures,
+      Map<String,Object> params) {
     super(name, features, norms, featureStoreName, allFeatures, params);
 
-    if (!hasParams()) {
-      throw new ModelException("Model " + name + " doesn't contain any weights");
-    }
+    final Map<String,Double> modelWeights = (params == null ? null
+        : (Map<String,Double>) params.get(WEIGHTS_PARAM));
 
-    final Map<String,Double> modelWeights = (Map<String,Double>) getParams()
-        .get(WEIGHTS_PARAM);
-    if ((modelWeights == null) || modelWeights.isEmpty()) {
-      throw new ModelException("Model " + name + " doesn't contain any weights");
+    featureToWeight = new Float[features.size()];
+
+    if (modelWeights != null) {
+      for (int i = 0; i < features.size(); ++i) {
+        final String key = features.get(i).getName();
+        final Double val = modelWeights.get(key);
+        featureToWeight[i] = (val == null ? null : new Float(val.floatValue()));
+      }
     }
+  }
 
-    // List<Feature> features = getFeatures(); // model features
-    featureToWeight = new float[features.size()];
+  @Override
+  public void validate() throws ModelException {
+    if (features.isEmpty()) {
+      // unusual but not an error
+      return;
+    }
 
+    int numFound = 0;
+    HashSet<String> missingWeightFeatureNames = new HashSet<String>();
     for (int i = 0; i < features.size(); ++i) {
-      final String key = features.get(i).getName();
-      if (!modelWeights.containsKey(key)) {
-        throw new ModelException("no weight for feature " + key);
+      if (featureToWeight[i] != null) {
+        ++numFound;
+      } else {
+        missingWeightFeatureNames.add(features.get(i).getName());
       }
-      featureToWeight[i] = modelWeights.get(key).floatValue();
+    }
+    if (numFound == 0) {
```

Comment:
quick possible simplification is use if(missingWeightFeatureNames.size() == features.size()) {
   throw new ModelException("Model " + name + " doesn't contain any weights");
}

and at the same time change if (featureToWeight[i] != null) {
to 

if (featureToWeight[i] == null)
missingWeightFeatureNames.add(features.get(i).getName());

this lets you reduce the number of lines of code and eliminate the numFound variable